### PR TITLE
LAC: also allow ConstraintMatrix::distribute for std::complex

### PR DESCRIPTION
**Pull request by tamiko**
Fix a typetrait check in ConstraintMatrix::distribute such that said class
and method can be used with complex valued vector types.

### CODE REVIEW

**Comment by tamiko** [comment]
A small regression introduced in 0bc683a79d8862978eea2d26e18b101d56d68b0f

File: include/deal.II/lac/constraint_matrix.templates.h [hunk 1]
```diff
@@ -852,12 +852,17 @@ ConstraintMatrix::distribute (VectorType &vec) const
...
   // the last else is for the simple case (sequential vector)
   const IndexSet vec_owned_elements = vec.locally_owned_elements();
+
+  // FIXME: This has to be refactored into a typetrait
   if ((typeid(vec) != typeid(Vector<double>)) &&
       (typeid(vec) != typeid(Vector<float>)) &&
+      (typeid(vec) != typeid(Vector<std::complex<double> >)) &&
       (typeid(vec) != typeid(BlockVector<double>)) &&
       (typeid(vec) != typeid(BlockVector<float>)) &&
+      (typeid(vec) != typeid(BlockVector<Vector<std::complex<double> > >)) &&
```
Comment (davydden) [inline include/deal.II/lac/constraint_matrix.templates.h]: probably you don't need `Vector` here.

File: tests/lac/constraint_matrix_distribute_complex.cc [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  GridGenerator::hyper_cube (triangulation);
+  DoFHandler<dim>      dof_handler(triangulation);
+  dof_handler.distribute_dofs (fe);
+
+  ConstraintMatrix constraints;
+  DoFTools::make_hanging_node_constraints (dof_handler,
+                                           constraints);
+  constraints.close ();
+
+  Vector<std::complex<double> > solution;
```
Comment (davydden) [inline tests/lac/constraint_matrix_distribute_complex.cc]: maybe do the same for `BlockVector` and `LA::Vector`?

File: include/deal.II/lac/constraint_matrix.templates.h [hunk 3]
```diff
@@ -852,12 +852,17 @@ ConstraintMatrix::distribute (VectorType &vec) const
   // call compress() finally. the first case here is for the complicated case,
   // the last else is for the simple case (sequential vector)
   const IndexSet vec_owned_elements = vec.locally_owned_elements();
+
+  // FIXME: This has to be refactored into a typetrait
```
Comment (bangerth) [inline include/deal.II/lac/constraint_matrix.templates.h]: open an issue for it.

**Comment by tamiko** [comment]
Updated

**Pull request merged**